tests/commands/run_test.py

```diff
@@ -334,3 +336,39 @@ def test_lots_of_files(mock_out_store_directory, tmpdir_factory):
             stderr=subprocess.STDOUT,
             env=env,
         )
+
+
+@pytest.mark.parametrize(
+    ('allow_unstaged_config',),
+    (
+        (True,),
+        (False,)
+    ),
+)
+def test_allow_unstaged_config_option(repo_with_passing_hook,
+                                      mock_out_store_directory,
+                                      allow_unstaged_config):
+
+    with cwd(repo_with_passing_hook):
+        with io.open(
+            '.pre-commit-config.yaml', 'a+',
+        ) as config_file:
+            # writing a newline should be relatively harmless to get a change
+            config_file.write('\n')
+
+    args = _get_opts(allow_unstaged_config=allow_unstaged_config)
+    ret, printed = _do_run(repo_with_passing_hook, args)
+    common_msg = 'You have an unstaged config file'
+    warning_msg = 'have specified the --allow-unstaged-config option.'
+
+    error_msg = 'have not specified the --allow-unstaged-config option.\n'
+
+    assert common_msg in printed
+    if allow_unstaged_config:
```

Comment:
I've found when there's two disparate branches in a test that it's often clearer as two separate tests

Here's a nice article I like to reference about the topic: http://googletesting.blogspot.com/2014/07/testing-on-toilet-dont-put-logic-in.html

In this case it might be a good idea to split the repeated bits into a helper and call that